app/api/companies.py

```diff
@@ -31,14 +32,40 @@ def revoke_token():
     pass
 
 
-@bp.route('/v1/companies/all', methods=['GET'])
+@bp.route('/v1/cities')
+def get_cities():
+
+    # Get the parameters from request
+    param_dict = request.args.to_dict()
+    # city_like = request.args.get('city_like')
+
+    # Check if city_like parameter is in GET request
+    if 'city_like' in param_dict:
+        city_like = '%' + param_dict['city_like'] + '%'
+
+        # Raw SQL query to get a selection of cities which fit the city_like parameter
+        query = db.session.execute(
+            "SELECT cities.city_name, COUNT(cities.city_id) AS city_count FROM COMPANIES JOIN CITIES ON companies.city_id=cities.city_id WHERE lower(cities.city_name) LIKE lower(:city_like) GROUP BY companies.city_id, cities.city_name", {"city_like": city_like})
+    else:
+        # Raw SQL query to get all cities with company count
+        query = db.session.execute(
+            "SELECT cities.city_name, COUNT(cities.city_id) AS city_count FROM COMPANIES JOIN CITIES ON companies.city_id=cities.city_id GROUP BY companies.city_id, cities.city_name")
+
+    # Returns a list with a list for every city
+    city_dict = dict((x, y) for x, y in query.fetchall())
+    sorted_list = sorted(city_dict.items(), key=lambda x: x[1], reverse=True)
```

Comment:
If we use a dict, we cannnot guarantee the order, if we use an array of tuples, we can. 

```json
[["Amsterdam", 160], ["Rotterdam", 51]]
```
If we do it in this way we can probably sort on a database level